src/Compilers/Core/Portable/SourceGenerator.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    public interface ISourceGenerator
```

Comment:
We discussed about this being an interface or class. Analyzers chose have a abstract base class, but we weren't sure what we'd end up a need since ultimately we only had one method on this. Keeping it an interface also was more natural since we have other interfaces that implement this interface as well for optional light-up.